source/wrapper/contracts/Wrapper.sol

```diff
@@ -74,42 +75,65 @@ contract Wrapper {
     require(order.signature.v != 0,
       "SIGNATURE_MUST_BE_SENT");
 
-    // The sender is sending ether that must be wrapped.
-    if (order.sender.token == address(wethContract)) {
+    // Check if sender is sending ether that must be wrapped.
+    _wrapEther(order.sender);
 
-      // Ensure message value is sender param.
-      require(order.sender.param == msg.value,
+    // Perform the swap.
+    swapContract.swap(order);
+
+    // Check if sender is receiving ether that must be unwrapped.
+    _unwrapEther(order.sender.wallet, order.signer.token, order.signer.param);
+  }
+
+  function provideDelegateOrder(
```

Comment:
/**
    * @notice Send an Order to a Delegate Contract
    * @dev 
    * @dev 
    * @param order Types.Order The Order,
    * @param delegate Delegate contract use to settle order
    */